MetaMask/metamask-extension
# Update @metamask/phishing-controller to v4.0.0

## Explanation

This updates the phishing controller to v4.0.0. The changelog for this can be found at the following URL: https://github.com/MetaMask/core/blob/main/packages/phishing-controller/CHANGELOG.md#400

The changes are as follows:
* Write a migration to account for the change in PhishingController state
* Update our phishing e2e tests to account for the new structure of data being returned from our phishing API 
* Update our metamask controller tests to account for the new structure of data being returned from our phishing API 

## Manual Testing Steps

* Visit test.metamask-phishing.io
* See that the page is successfully blocked. 
* Click `back to safety` and see that you are moved away from the site (on the chrome release, that seems to be back to the MetaMask wallet)
* Visit the page again but this time click `continue to site`.
* See that you are brought to the site

## Pre-merge author checklist


## Pre-merge reviewer checklist


If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.
